### ISSUE TITLE
Jittery Video in Firefox 119

### ISSUE THREAD

**Issue by TheMegamind** [mrlt8/docker-wyze-bridge#1025]
This is most likely a bug in Firefox, but I'm posting here because I've only encountered this with Wyze-Bridge and was curious if others have seen similar behavior.

Starting very recently, coinciding I believe with the Firefox 119 update, the live video from all of my cameras (from a V2 Pan to as V3 and even a V3 Pro) has become jittery as per the video below. The time stamp jumps around and occasionally (though not in this video) the window flashes green for a fraction of a second).

I tested with both Chrome and Edge and the video was smooth and unaffected. So, going back to FF, I tested with another existing user profile as well a brand new, wholly untouched profile and the jitter issue was present with both. With further tinkering, I observed that disabling Hardware Acceleration in Firefox under Settings → General → Performance and restarting the browser eliminated the issue. 
 
https://github.com/mrlt8/docker-wyze-bridge/assets/12296628/927a3998-f721-4ddb-aca3-d1f09664a03f

Windows 11 w/NVIDIA GeForce GTS 1050 Ti

**Comment by mrlt8** [comment]
Interesting. Could be related to the way the video is encoded by the cameras which is causing issues with hardware decoding.

**Comment by TheMegamind** [comment]
It's only become an issue on with Firefox with release 118 or 119, and (so far at least) with Wyze-Bridge. I do not have the same issue with Hardware Acceleration enabled in Edge and Chrome. It's very frustrating. I guess I'll wait and see if there's a fix in FF 120, then take it from there.

**Comment by d-allin** [comment]
Man, do I ever feel dumb now. I've spent hours troubleshooting this, assuming it was a problem with the bridge or just wyze being wyze. Of course, I never thought to check the playback outside of Firefox.

Looks like this is moving on bugzilla.
[
NVENC-encoded H.264 sent over WebRTC to Firefox 119 on Windows results in a glitching stream - a problem which doesn't exist in Firefox 118 and 120.0beta.
](https://bugzilla.mozilla.org/show_bug.cgi?id=1861895)

**Comment by TheMegamind** [comment]
@d-allin Yeah, I spent a lot of time troubleshooting this as well before posting here. Thanks for the bugzilla.  Maybe I'll post a link back to here just in case it might be useful to someone.

**Comment by peinchka** [comment]
I've posted some quick-fixes for this issue at the Bugzilla link. Hopefully, I've also now provided enough info to Mozilla to quickly and fully resolve this problem in Firefox 110, 119 and 120.

**Comment by TheMegamind** [comment]
@mrlt8 As Firefox has given this bug low priority, is there any way to implement an optional temporarily fix for those of us affected? I know it's not your issue, but I've had to deal the jittery video for too long now. 

> For anyone else experiencing this issue, if you’re able to modify the HTML code of the webpage, slightly blurring a single pixel has been found to completely remedy the glitching problem in all affected Firefox versions. A line like this is all you need:

`<div style="width: 1px; height: 1px; position: fixed; backdrop-filter: blur(0.01rem)"></div>`

**Comment by mrlt8** [comment]
@TheMegamind Can that go anywhere in the page or do we need one near each video tag?

**Comment by TheMegamind** [comment]
Good question. I'm not sure since it's not my solution, but rather was borrowed from the [bugzilla link](https://bugzilla.mozilla.org/show_bug.cgi?id=1861895). There are a couple of references to that and similar workarounds that might be helpful. 

I see the commit. Is there an easy for me to test?

**Comment by mrlt8** [comment]
Changes should be in the dev branch which you can pull using the dev image `mrlt8/wyze-bridge:dev`.

**Comment by TheMegamind** [comment]
Okay, that works in multi-camera view, but not for a single camera full-screen.

**Comment by mrlt8** [comment]
Single camera full-screen page is part of [mediamtx](https://github.com/bluenviron/mediamtx), so we'd need to do a PR to the html that mtx generates.

**Comment by TheMegamind** [comment]
Given that this a bug in Firefox, not mediamtx, is that something one can reasonably do?  It feels like that might be pushing things, but either way, I don't have a working knowledge of mediamtx or feel sufficiently qualified to submit a proper PR. I'll just have to follow your lead on what's best.  At least the latest commit above gives us a partial workaround for now, so thank you for that much!

**Comment by mrlt8** [comment]
I modified the fix to use CSS so that the pixel only displays in firefox. Can you pull a fresh dev image to test?

**Comment by TheMegamind** [comment]
Pulled the dev image and it is working as expected with both webRTC and HLS.

**Comment by TheMegamind** [comment]
@mrlt8 This issue appears to have been resolved in Firefox 123 via [1817617](https://bugzilla.mozilla.org/show_bug.cgi?id=1817617).
